src/Centreon/Domain/Authentication/UseCase/AuthenticateApi.php

```diff
@@ -165,53 +187,32 @@ private function getUserFromProviderOrFail(ProviderInterface $localProvider): Co
     }
 
     /**
-     * Create the authentication token or throw an Exception.
-     *
-     * @param string $token
-     * @param ProviderInterface $localProvider
-     * @param ContactInterface $contact
-     * @return void
+     * @inheritDoc
      */
-    private function createApiAuthenticationTokenOrFail(
+    public function createAPIAuthenticationTokens(
         string $token,
-        ProviderInterface $localProvider,
-        ContactInterface $contact
-    ): void {
-        /**
-         * Create the token.
-         */
-        $this->authenticationService->createAPIAuthenticationTokens(
-            $token,
-            $localProvider->getConfiguration(),
-            $contact,
-            $localProvider->getProviderToken($token),
-            null
-        );
-    }
-
-    /**
-     * Set the authentication to the response.
-     *
-     * @param AuthenticateApiResponse $response
-     * @param ContactInterface $contact
-     * @param string $token
-     */
-    private function setResponseAuthentication(
-        AuthenticateApiResponse $response,
+        ProviderConfiguration $providerConfiguration,
         ContactInterface $contact,
-        string $token
+        ProviderToken $providerToken,
+        ?ProviderToken $providerRefreshToken
     ): void {
-        /**
-         * Prepare the response with contact informations and API authentication token.
-         */
-        $response->setApiAuthentication($contact, $token);
         $this->debug(
-            "[AUTHENTICATE API] Authentication success",
-            [
-                "provider_name" => LocalProvider::NAME,
-                "contact_id" => $contact->getId(),
-                "contact_alias" => $contact->getAlias()
-            ]
+            '[AUTHENTICATION SERVICE] Creating authentication tokens for user',
+            ['user' => $contact->getAlias()]
         );
+        try {
+            if ($providerConfiguration->getId() === null) {
+                throw new \InvalidArgumentException("Provider configuration can't be null");
+            }
```

Comment:
Move before this try/catch
```suggestion
            if ($providerConfiguration->getId() === null) {
                throw new \InvalidArgumentException("Provider configuration can't be null");
            }
```